--- php-8.2.6+wago1/configure.ac.orig	2023-05-15 15:45:10.135626994 +0200
+++ php-8.2.6+wago1/configure.ac	2022-05-15 15:45:11.575847368 +0200
@@ -366,7 +366,7 @@
 PHP_CHECK_FUNC(gethostbyaddr, nsl, network)
 PHP_CHECK_FUNC(dlopen, dl, root)
 PHP_CHECK_FUNC(dlsym, dl, root)
-if test "$ac_cv_func_dlopen" = "yes"; then
+if test "$ac_cv_func_dlopen" = "yes" || test "$ac_cv_lib_dl_dlopen" = "yes"; then
   AC_DEFINE(HAVE_LIBDL, 1, [ ])
 fi
 AC_CHECK_LIB(m, sin)
