From: Oleg Ganin <oleg.ganin@wago.com>
Date: Tue, 2 Apr 2024 16:25:56 +0200
Subject: [PATCH] Patch not compatible compiler flag

---
 configure | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/configure b/configure
index f43a58af5ba2..c1435fcf6541 100755
--- a/configure
+++ b/configure
@@ -9565,12 +9565,12 @@ rm -f core conftest.err conftest.$ac_objext conftest.beam \
 	# flag at all (https://bugzilla.mindrot.org/show_bug.cgi?id=3629)
 	case "$CLANG_VER" in
 	apple-15*) {
-	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -fzero-call-used-regs=used and linking succeeds" >&5
-printf %s "checking if $CC supports compile flag -fzero-call-used-regs=used and linking succeeds... " >&6; }
+	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -fzero-call-used-regs=skip and linking succeeds" >&5
+printf %s "checking if $CC supports compile flag -fzero-call-used-regs=skip and linking succeeds... " >&6; }
 	saved_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS $WERROR -fzero-call-used-regs=used"
+	CFLAGS="$CFLAGS $WERROR -fzero-call-used-regs=skip"
 	_define_flag=""
-	test "x$_define_flag" = "x" && _define_flag="-fzero-call-used-regs=used"
+	test "x$_define_flag" = "x" && _define_flag="-fzero-call-used-regs=skip"
 	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -9716,12 +9716,12 @@ rm -f core conftest.err conftest.$ac_objext conftest.beam \
 } ;;
 	17*)	;;
 	*)	{
-	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -fzero-call-used-regs=used and linking succeeds" >&5
-printf %s "checking if $CC supports compile flag -fzero-call-used-regs=used and linking succeeds... " >&6; }
+	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if $CC supports compile flag -fzero-call-used-regs=skip and linking succeeds" >&5
+printf %s "checking if $CC supports compile flag -fzero-call-used-regs=skip and linking succeeds... " >&6; }
 	saved_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS $WERROR -fzero-call-used-regs=used"
+	CFLAGS="$CFLAGS $WERROR -fzero-call-used-regs=skip"
 	_define_flag=""
-	test "x$_define_flag" = "x" && _define_flag="-fzero-call-used-regs=used"
+	test "x$_define_flag" = "x" && _define_flag="-fzero-call-used-regs=skip"
 	cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
