From: Florian Seidel <florian.seidel@wago.com>
Date: Wed, 26 Jul 2023 10:43:36 +0200
Subject: [PATCH] chpass_pam_talker: fix broken conversation function

---
 pam.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/pam.c b/pam.c
index 438549a33db2..fd71e83703ed 100644
--- a/pam.c
+++ b/pam.c
@@ -153,9 +153,21 @@ int chpass_pam_talker(int num_msg,
 			case PAM_PROMPT_ECHO_OFF:
 				response[i].resp = strdup((userinfo->count++) ? userinfo->newpw : userinfo->oldpw);
 				break;
+			case PAM_ERROR_MSG: // fall through
+			case PAM_TEXT_INFO:
+				// ignore, not an error
+				response[i].resp_retcode = PAM_SUCCESS;
+				response[i].resp = NULL;
+				break;
 			default:
-				_pam_drop_reply(response, i);
-				return PAM_CONV_ERR;
+				response[i].resp_retcode = PAM_CONV_ERR;
+				response[i].resp_retcode = NULL;
+				// _pam_drop_reply invalidates _all_ requests (macro
+				// name is missleading) and then an error is
+				// returned. Thats why we getting a conversation
+				// error in the log.
+				//_pam_drop_reply(response, i);
+				//return PAM_CONV_ERR;
 		}
 	}
 	/* everything okay, set PAM response values */
