From: "Ristok, Anatolij" <ark@mm-software.com>
Date: Tue, 14 May 2024 12:31:40 +0200
Subject: [PATCH] Add additional suppressions not covered by glib supplied
 glib.supp

---
 glib.supp | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/glib.supp b/glib.supp
index a732aa512d4f..6cf2fff29449 100644
--- a/glib.supp
+++ b/glib.supp
@@ -957,3 +957,33 @@
 	...
 	fun:g_file_*
 }
+
+{
+   wago-glib-leak-g_quark_init
+   Memcheck:Leak
+   match-leak-kinds: reachable,possible
+   ...
+   fun:g_quark_init
+   ...
+   fun:_dl_init
+}
+
+{
+   wago-glib-leak-_g_signal_init
+   Memcheck:Leak
+   match-leak-kinds: reachable,possible
+   ...
+   fun:_g_signal_init
+   ...
+   fun:_dl_init
+}
+
+{
+   wago-glib-leak-gobject_init_ctor
+   Memcheck:Leak
+   match-leak-kinds: reachable,possible
+   ...
+   fun:gobject_init_ctor
+   ...
+   fun:_dl_init
+}
\ No newline at end of file
