From 9ccd9fad75b85b35fd4a34fbb51142ff887df563 Mon Sep 17 00:00:00 2001
From: Falk Werner <falk.werner@gmx.net>
Date: Thu, 17 Nov 2022 18:20:22 +0100
Subject: [PATCH] introduce "strong" option to allow duplicate file matches

---
 config.c       | 39 ++++++++++++++++++++++++++-------------
 logrotate.8.in |  4 ++++
 logrotate.h    |  1 +
 3 files changed, 31 insertions(+), 13 deletions(-)

diff --git a/config.c b/config.c
index 46f2b14e..0e539f78 100644
--- a/config.c
+++ b/config.c
@@ -1433,6 +1433,8 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                         newlog->flags |= LOG_FLAG_MISSINGOK;
                     } else if (!strcmp(key, "nomissingok")) {
                         newlog->flags &= ~LOG_FLAG_MISSINGOK;
+                    } else if (!strcmp(key, "strong")) {
+                        newlog->flags |= LOG_FLAG_STRONG;
                     } else if (!strcmp(key, "prerotate")) {
                         freeLogItem (pre);
                         scriptStart = start;
@@ -1849,6 +1851,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                         for (glob_count = 0; glob_count < globResult.gl_pathc; glob_count++) {
                             struct logInfo *log;
                             struct stat sb_glob;
+                            int add_file = 1;
 
                             /* if we glob directories we can get false matches */
                             if (!lstat(globResult.gl_pathv[glob_count], &sb_glob) &&
@@ -1862,24 +1865,34 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                                 for (k = 0; k < log->numFiles; k++) {
                                     if (!strcmp(log->files[k],
                                                 globResult.gl_pathv[glob_count])) {
-                                        message(MESS_ERROR,
-                                                "%s:%d duplicate log entry for %s\n",
-                                                configFile, lineNum,
-                                                globResult.gl_pathv[glob_count]);
-                                        logerror = 1;
-                                        goto duperror;
+                                        if (log->flags & LOG_FLAG_STRONG) {
+                                            add_file = 0;
+                                            message(MESS_DEBUG,
+                                                    "%s:%d ignore duplicate log entry for %s\n",
+                                                    configFile, lineNum,
+                                                    globResult.gl_pathv[glob_count]);
+                                        } else {
+                                            message(MESS_ERROR,
+                                                    "%s:%d duplicate log entry for %s\n",
+                                                    configFile, lineNum,
+                                                    globResult.gl_pathv[glob_count]);
+                                            logerror = 1;
+                                            goto duperror;
+                                        }
                                     }
                                 }
                             }
 
-                            newlog->files[newlog->numFiles] =
-                                strdup(globResult.gl_pathv[glob_count]);
-                            if (newlog->files[newlog->numFiles] == NULL) {
-                                message_OOM();
-                                logerror = 1;
-                                goto duperror;
+                            if (add_file) {
+                                newlog->files[newlog->numFiles] =
+                                    strdup(globResult.gl_pathv[glob_count]);
+                                if (newlog->files[newlog->numFiles] == NULL) {
+                                    message_OOM();
+                                    logerror = 1;
+                                    goto duperror;
+                                }
+                                newlog->numFiles++;
                             }
-                            newlog->numFiles++;
                         }
 duperror:
                         globfree(&globResult);
diff --git a/logrotate.8.in b/logrotate.8.in
index 923178d9..428da57f 100644
--- a/logrotate.8.in
+++ b/logrotate.8.in
@@ -312,6 +312,10 @@ message.  See also \fBnomissingok\fR.
 \fBnomissingok\fR
 If a log file does not exist, issue an error.  This is the default.
 
+.TP
+\fBstrong\fR
+Ignore any following matches of a log file.
+
 .TP
 \fBifempty\fR
 Rotate the log file even if it is empty, overriding the \fBnotifempty\fR
diff --git a/logrotate.h b/logrotate.h
index ff82e900..545ec461 100644
--- a/logrotate.h
+++ b/logrotate.h
@@ -28,6 +28,7 @@
 #define LOG_FLAG_TMPFILENAME    (1U << 14)
 #define LOG_FLAG_DATEHOURAGO    (1U << 15)
 #define LOG_FLAG_ALLOWHARDLINK  (1U << 16)
+#define LOG_FLAG_STRONG         (1U << 17)
 
 #define NO_MODE ((mode_t) -1)
 #define NO_UID  ((uid_t) -1)
