Index: containerd-1.6.9/Makefile
===================================================================
--- containerd-1.6.9.orig/Makefile
+++ containerd-1.6.9/Makefile
@@ -31,7 +31,7 @@ TEST_IMAGE_LIST ?=
 
 # Used to populate variables in version package.
 VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
 PACKAGE=github.com/containerd/containerd
 SHIM_CGO_ENABLED ?= 0
 
