From: Bill Fenner <fenner@gmail.com>
Date: Fri, 25 Nov 2022 08:41:46 -0800
Subject: [PATCH] apps: snmpset: allow SET with NULL varbind for testing

---
 apps/snmpset.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apps/snmpset.c b/apps/snmpset.c
index 48e14bd5130d..d542713e1ba5 100644
--- a/apps/snmpset.c
+++ b/apps/snmpset.c
@@ -182,6 +182,7 @@ main(int argc, char *argv[])
             case 'x':
             case 'd':
             case 'b':
+            case 'n': /* undocumented */
 #ifdef NETSNMP_WITH_OPAQUE_SPECIAL_TYPES
             case 'I':
             case 'U':
