--- sssd-2.5.2/src/external/ldap.m4	2021-07-12 20:53:56.000000000 +0200
+++ sssd-2.5.2/src/external/ldap.m4	2021-09-07 14:41:15.031034692 +0200
@@ -69,7 +69,7 @@
                 ldap_parse_derefresponse_control \
                 ldap_derefresponse_free])
 AC_CHECK_MEMBERS([struct ldap_conncb.lc_arg],
-                 [AC_RUN_IFELSE(
+                 [AC_COMPILE_IFELSE(
                    [AC_LANG_PROGRAM(
                      [[ #include <ldap.h> ]],
                      [[
@@ -78,8 +78,7 @@
                      ]] )],
                    [AC_DEFINE([HAVE_LDAP_CONNCB], [1],
                      [Define if LDAP connection callbacks are available])],
-                   [AC_MSG_WARN([Found broken callback implementation])],
-                   [])],
+                   [AC_MSG_WARN([Found broken callback implementation])] )],
                  [], [[#include <ldap.h>]])
 
 AC_CHECK_TYPE([LDAPDerefRes],
