Index: azure-iot-sdk-c-1.6.0/c-utility/src/hmac.c
===================================================================
--- azure-iot-sdk-c-1.6.0.orig/c-utility/src/hmac.c
+++ azure-iot-sdk-c-1.6.0/c-utility/src/hmac.c
@@ -208,7 +208,7 @@ int hmacFinalBits(HMACContext *ctx,
 *   sha Error Code.
 *
 */
-int hmacResult(HMACContext *ctx, uint8_t *digest)
+int hmacResult(HMACContext *ctx, uint8_t digest[USHAMaxHashSize])
 {
     if (!ctx) return shaNull;
 
